src/EventListener/ReadRateLimitAnnotationListener.php

```diff
@@ -61,24 +65,48 @@ public function onKernelController(ControllerEvent $event): void
             }
         }
         $reflection = new \ReflectionClass($controllerName);
-        $annotation = $this->annotationReader->getMethodAnnotation($reflection->getMethod((string) ($methodName ?? '__invoke')), RateLimitAnnotation::class);
+        $annotation = $this->annotationReader->getMethodAnnotation($reflection->getMethod((string) ($methodName ?? '__invoke')), RateLimitAnnotation::class) ?? $this->annotationReader->getMethodAnnotation($reflection->getMethod((string) ($methodName ?? '__invoke')), GraphQLRateLimitAnnotation::class);
 
-        if (!$annotation instanceof RateLimitAnnotation) {
+        if (!$annotation instanceof RateLimitAnnotation && !$annotation instanceof GraphQLRateLimitAnnotation) {
             return;
         }
 
-        if ($this->limitByRoute) {
-            $rateLimit = new RateLimit(
-                $annotation->getLimit() ?? $this->limit,
-                $annotation->getPeriod() ?? $this->period
-            );
-
-            $rateLimit->varyHashOn('_route', $request->attributes->get('_route'));
-        } else {
+        if ($annotation instanceof RateLimitAnnotation) {
             $rateLimit = new RateLimit(
                 $this->limit,
                 $this->period
             );
+
+            if ($this->limitByRoute) {
+                $rateLimit = new RateLimit(
+                    $annotation->getLimit() ?? $this->limit,
+                    $annotation->getPeriod() ?? $this->period
+                );
+
+                $rateLimit->varyHashOn('_route', $request->attributes->get('_route'));
+            }
+        }
+
+        if ($annotation instanceof GraphQLRateLimitAnnotation) {
```

Comment:
To me this listener has a lot of code, what do you think about create a new for GraphQLRateLimiteAnnotation?